### PR TITLE
Sub/Copter: changed log messages to enum 

### PR DESCRIPTION
**Pull request by EbinPhilip**
Fix for issue #7839 
-  Changed all the log message IDs in define.h in copter/sub to enum. 
- Removed Duplicate IDs

### CODE REVIEW

**Comment by OXINARF** [comment]
This PR is a duplicate of #7895. I'll leave it open for now since the other one also needs to be fixed. If the other one isn't fixed after some time, I'll look at this one for merging.

@EbinPhilip Have you read our contribution rules (http://ardupilot.org/dev/docs/submitting-patches-back-to-master.html)? Most important ones are the commit message format, one commit per-vehicle/library and cleaning the commit history with rebase and squash.

**Comment by EbinPhilip** [comment]
@OXINARF I have removed those commits and made new ones separately for Copter and Sub, as per the guidelines. Is this fine? 
The code did build successfully for me. I think the checks failed due to some external reasons. Can you please run them again?

**Comment by rmackay9** [comment]
One small thing is I don't think we need the "= 0", etc in the enum because the actual message number used isn't important.  The "FMT" messages tell the ground station what's in each message.

File: ArduCopter/defines.h [hunk 1]
```diff
@@ -312,29 +312,31 @@ enum DevOptions {
...
-#define LOG_GUIDEDTARGET_MSG            0x22
-#define LOG_THROW_MSG                   0x23
+enum LoggingParameters {
+     TYPE_AIRSTART_MSG,
+     TYPE_GROUNDSTART_MSG,
+     LOG_CONTROL_TUNING_MSG,
+     LOG_NAV_TUNING_MSG,
+     LOG_OPTFLOW_MSG,
+     LOG_EVENT_MSG,
+     LOG_PID_MSG,                 // deprecated
```
Comment (WickedShell) [inline ArduCopter/defines.h]: Logging parameters are actually different then MAVLink/autopilot paramters. The enum value doesn't strictly matter between versions. What this means is that these deprecated ones can just be removed directly if nothing is using it, which leaves the ID's clearly available for usage in the future.

**Comment by EbinPhilip** [comment]
@rmackay9 I have removes the enum values @WickedShell I did that. Tried on SITL, worked fine. But Travis fails for SITL copter, so i undid it. I'm thinking, maybe travis is using those parameters. What should i do?

**Comment by WickedShell** [comment]
@EbinPhilip it may have been just a CI failure (sitl tests will sometimes fail for other reasons). Can you push the commit back up so someone can see the log of it failing? I can't seem to remember if there is a way to access an old travis log.

**Comment by EbinPhilip** [comment]
@WickedShell Here is the log: https://api.travis-ci.org/v3/job/352657770/log.txt
Can you restart the travis test?

**Comment by WickedShell** [comment]
@EbinPhilip seems to be running now. I only looked briefly but the error looks like an unassociated SITL failure. (I saw the same style error earlier today)

**Comment by EbinPhilip** [comment]
@WickedShell Yup. Just some PR unrelated gliche. Any other changes to be made?

**Comment by OXINARF** [comment]
@EbinPhilip The only thing left is squashing everything up, one commit for Copter and another for Sub is enough - you can write in the commit sub-message that deprecated ids and numbers were removed.

@rmackay9 Indeed, we don't need the numbers, but we need vehicle IDs to be between 0 and 64, because the general DataFlash ones start with 64.

**Comment by EbinPhilip** [comment]
@OXINARF I have done that. Anything more to be done?

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request closed without merging**